Source/JavaScriptCore/dfg/DFGByteCodeParser.cpp

```diff
@@ -7830,174 +8013,22 @@ void ByteCodeParser::parseBlock(unsigned limit)
-                        break;
-                    }
-                }
-                
-                SpeculatedType prediction = getPrediction();
-                NodeType nodeType;
-                if (resolveType == GlobalVar || resolveType == GlobalVarWithVarInjectionChecks)
-                    nodeType = GetGlobalVar;
-                else
-                    nodeType = GetGlobalLexicalVariable;
-                Node* value = addToGraph(nodeType, OpInfo(operand), OpInfo(prediction));
-                if (resolveType == GlobalLexicalVar || resolveType == GlobalLexicalVarWithVarInjectionChecks)
-                    addToGraph(CheckNotEmpty, value);
-                set(bytecode.m_dst, value);
-                break;
-            }
-            case ResolvedClosureVar:
-            case ClosureVar:
-            case ClosureVarWithVarInjectionChecks: {
-                Node* scopeNode = get(bytecode.m_scope);
-                
-                // Ideally we wouldn't have to do this Phantom. But:
-                //
-                // For the constant case: we must do it because otherwise we would have no way of knowing
-                // that the scope is live at OSR here.
-                //
-                // For the non-constant case: GetClosureVar could be DCE'd, but baseline's implementation
-                // won't be able to handle an Undefined scope.
-                addToGraph(Phantom, scopeNode);
-                
-                // Constant folding in the bytecode parser is important for performance. This may not
-                // have executed yet. If it hasn't, then we won't have a prediction. Lacking a
-                // prediction, we'd otherwise think that it has to exit. Then when it did execute, we
-                // would recompile. But if we can fold it here, we avoid the exit.
-                if (JSValue value = m_graph.tryGetConstantClosureVar(scopeNode, ScopeOffset(operand))) {
-                    set(bytecode.m_dst, weakJSConstant(value));
-                    break;
-                }
-                SpeculatedType prediction = getPrediction();
-                set(bytecode.m_dst,
-                    addToGraph(GetClosureVar, OpInfo(operand), OpInfo(prediction), scopeNode));
-                break;
-            }
-            case UnresolvedProperty:
-            case UnresolvedPropertyWithVarInjectionChecks:
-            case ModuleVar:
-            case Dynamic:
-                RELEASE_ASSERT_NOT_REACHED();
-                break;
-            }
+            getFromScopeHelper(bytecode, bytecode.m_scope);
             NEXT_OPCODE(op_get_from_scope);
         }
 
+        case op_resolve_and_get_from_scope: {
+            auto bytecode = currentInstruction->as<OpResolveAndGetFromScope>();
+            auto setResolvedScope = [&] (VirtualRegister resolvedScope, Node* value) {
+                set(resolvedScope, value);
+                set(Operand::tmp(bytecode.tmpResolvedScope), value);
```

Comment:
Why do we set two things here?

Reply:
tmpResolvedScope is used to store the resolved scope for OSR exit.